content/docs/atom-guide/look-dev/tools/material-canvas.md

```diff
@@ -0,0 +1,302 @@
+            }
+        ]
+    }
+}
+```
+
+#### Material Graph Node Configuration Attributes
+Every node configuration must have a unique ID. This ensures that they are uniquely identifiable, regardless of location on disk, project, name collisions, or other factors.
+
+The node category determines how nodes will be grouped together in the node palette tree.
+
+The node title is the name displayed in the node palette and on the top of the node in the graph view. It will also be used to help create unique symbol names and variable names in shader code.
+
+The node title palette name is an optional field specifying which style sheet palette to use for the title bar on the node. Style sheets configure styling, coloring, fonts, and other attributes that control how elements in the node palette and graph view are displayed. Style sheets are defined in a separate, application wide file.
+
+The node description is an optional field that will be presented as a tool tip when hovering over a node in the palette.
+
+Slot data type groups contains a delimited list of slots names. The Material Canvas graph traversal and code generation process will enforce that all slot names listed in this field are promoted to the same data type, if they are compatible. Currently, if all of the listed slots reference scalar or vector values than all of the slot values will be promoted to the largest data type. For example, if all of the slots on the node are scalar values but an incoming connection is a three-dimensional vector then all of the other slots will be up converted to three dimensional vectors. This is necessary, in addition to other forms of casting, so that variables generated from different incoming types will be compatible with code and function calls defined in the node.
+
+#### Material Graph Node Slot Configuration Attributes
+Every slot configuration must have a unique name with respect to the node. The slot name is used to uniquely identify and address the slot, setup connections between slots, and is used to create a unique variable name in generated shader code and other files. The name will also be used as the display name in the UI if there is no specific display name specified. Changing the name will break connections and lose any data associated with the slot.
+
+The display name can be specified as a more user-friendly name to present in the UI. If no note display name is assigned then one will be inferred from the slot name.
+
+The description provides more detail about what slots do or represent. These will be presented as tool tips when hovering the mouse cursor over the node and slot in the graph view.
+
+The supporting data types regular expression field is used to acquire which data types are compatible with a slot. The regular expression makes it easier to match several data types with a single expression instead of listing them all individually. However, individual data types can be listed using vertical bar separators between them.
```

Comment:
Should there be an apostrophe here? 
Reworded some of this and linked to Regex wiki. 
```suggestion
The supporting data types' regular expression field is used to acquire which data types are compatible with a slot. With [regular expressions](https://en.m.wikipedia.org/wiki/Regular_expression), you can query multiple data types that match a specific pattern, or list the data types individually. 
```